types/d3-dispatch/index.d.ts

```diff
@@ -1,6 +1,27 @@
 // Last module patch version validated against: 3.0.1
 
-export interface Dispatch<T extends object> {
+export namespace Dispatch {
+    /** given a string like `a.1 b c.2`, it returns a union like `'a' | 'b' | 'c'` */
+    type ExtractEventNames<Input extends string> = Input extends "" ? never
```

Comment:
These helper types will end up exported; was that intentional?

Reply:
it was somewhat intentional because of the phrasing of [`@definitelytyped/strict-export-declare-modifiers`](https://github.com/microsoft/DefinitelyTyped-tools/blob/main/packages/eslint-plugin/src/rules/strict-export-declare-modifiers.ts):

> _[...] If you have a good reason not to export this declaration, add `export {}` [...]_

Anyway, shall I make these first 2 helpers private instead? I doubt any consumers will need them.

Reply:
actually we need to access `ExtractEventNames` in the unit tests. So that one must be exported

Reply:
Export as much as you feel makes sense, I'm just checking since if it's added it'll be there for good.